helper/logging/logging.go

```diff
@@ -92,6 +97,34 @@ func CurrentLogLevel() string {
 	return logLevel
 }
 
+// WhitelistedLogPatterns returns a list of whitelisted log line patterns.
+// Matching lines will be logged regardless of the log level.
+func WhitelistedLogPatterns() []*regexp.Regexp {
+	patterns := make([]*regexp.Regexp, 0, 0)
+
+	environ := os.Environ()
+
+	for _, e := range environ {
+		if !strings.HasPrefix(e, EnvLogPatternPrefix) {
+			continue
+		}
+
+		splits := strings.SplitN(e, "=", 2)
+		envVarName := splits[0]
+		envVarValue := splits[1]
+
+		pattern, err := regexp.Compile(envVarValue)
+
+		if err != nil {
+			log.Fatalln("Can not compile "+envVarName+":", err)
```

Comment:
I am not sure if it is ok to panic if the pattern can not compile. But I guess the user expects terraform to fail if this is the case.